src/extension/provider/annotations.ts

```diff
@@ -0,0 +1,64 @@
+import {
+  window,
+  commands,
+  NotebookCell,
+  NotebookCellOutput,
+  NotebookCellOutputItem,
+  NotebookCellStatusBarItemProvider,
+  NotebookCellStatusBarItem,
+  NotebookCellStatusBarAlignment,
+} from 'vscode'
+
+import { OutputType } from '../../constants'
+import { CellOutputPayload } from '../../types'
+import { Kernel } from '../kernel'
+import { getAnnotations } from '../utils'
+
+export class AnnotationsProvider implements NotebookCellStatusBarItemProvider {
+  constructor(private readonly kernel: Kernel) {
+    commands.registerCommand(
+      'runme.openCellAnnotations',
+      async (cell: NotebookCell) => {
+        try {
+          const exec = await kernel.createCellExecution(cell)
+          exec.start(Date.now())
```

Comment:
Can we encapsulate the start as part of createCellExecution, or there is any other use cases when we need more control in order to set the ellapsed time for cell execution?

Reply:
I see your point. I generally think this code needs a refactor. It feels like a bad code smell to me to rely on `kernel` being a closure here. However, this is the first time we're using a Notebook Output to provide a UI so I'd like to hold off with generalization until we have a better handle on it. However, your point about encapsulating impl detail here is well taken.